components/arm/arm.go

```diff
@@ -153,7 +153,19 @@ func Plan(ctx context.Context, logger golog.Logger, a Arm, dst spatialmath.Pose)
 	if err != nil {
 		return nil, err
 	}
-	return motionplan.PlanFrameMotion(ctx, logger, dst, model, model.InputFromProtobuf(jp), defaultArmPlannerOptions, nil)
+	planRequest := motionplan.PlanRequest{
+		Logger:             logger,
+		Goal:               referenceframe.NewPoseInFrame(referenceframe.World, dst),
```

Comment:
Stuff like this feels off to me, and is a reason that I don't love deprecating `PlanFrameMotion`. We're creating a PoseInFrame, independently of any framesystem, which doesn't really make sense. It only functions because of special-case behavior in another package. It makes me think we have the wrong abstraction somewhere. Curious what others think.

Reply:
Yeah I agree with you. We had talked in a meeting about not deprecating this but I don't think it ever made its way into the ticket

Reply:
> We're creating a PoseInFrame, independently of any framesystem, which doesn't really make sense. It only functions because of special-case behavior in another package. It makes me think we have the wrong abstraction somewhere.

I agree with this. I will add back `PlanFrameMotion`.